### ISSUE TITLE
Flood of error messages when connection closes

### ISSUE THREAD

**Issue by sander2** [paritytech/jsonrpsee#157]
When the connection is closed (because e.g. the server shuts down), there is an unending flood of these messages:
```
ERROR jsonrpsee::client] Client Error: Inner(Ws(Closed))
```

It would be nice if this error was handled a bit more graciously.

**Comment by niklasad1** [comment]
Indeed, it's on the checklist for releasing `v2` also described in #67 

Currently, when an error is received it just logged and not propagated to the user because in most cases it's not possible to determine which request failed because it's mapped per `request id`, as in the JSONRPC v2 spec.

Thus, I think the proper way is just to terminate the connection when an error is received on the connection (with the error to user)

Thoughts?

**Comment by sander2** [comment]
In the case of an unrecoverable error like a closed connection I think it should definitely be propagated to the user. I am not sure what should happen when an error occurs that is only effecting one request, but just ignoring it probably isn't a good idea, as I imagine it could lead to unexpected program freezes

**Comment by alannotnerd** [comment]
+1
